File: api-tester/src/defaults/java/com/revenuecat/apitester/java/revenuecatui/PaywallColorAPI.java
package com.revenuecat.apitester.java.revenuecatui;

import android.graphics.Color;

import com.revenuecat.purchases.paywalls.PaywallColor;

@SuppressWarnings({"unused"})
final class PaywallColorAPI {
    static void check(PaywallColor paywallColor) {
        String stringRepresentation = paywallColor.getStringRepresentation();
        Color underlyingColor = paywallColor.getUnderlyingColor();
        Integer colorInt = paywallColor.getColorInt();
    }

    static void checkConstructor(Integer colorInt) {
        PaywallColor paywallColor = new PaywallColor("#FFFFFF");
        PaywallColor paywallColor2 = new PaywallColor(colorInt);
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/revenuecatui/PaywallColorAPI.kt
package com.revenuecat.apitester.kotlin.revenuecatui

import android.graphics.Color
import android.os.Build
import androidx.annotation.RequiresApi
import com.revenuecat.purchases.paywalls.PaywallColor

@Suppress("unused", "UNUSED_VARIABLE")
@RequiresApi(Build.VERSION_CODES.O)
private class PaywallColorAPI {
    fun check(paywallColor: PaywallColor) {
        val stringRepresentation: String = paywallColor.stringRepresentation
        val underlyingColor: Color? = paywallColor.underlyingColor
        val colorInt: Int = paywallColor.colorInt
    }

    fun checkConstructor(colorInt: Int) {
        val paywallColor = PaywallColor("#FFFFFF")
        val paywallColor2 = PaywallColor(colorInt)
    }
}
